frontend/util/dropdown-menu-modal.component.tsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import { useCss } from "kremling";
+import kabobIcon from "../../icons/148705-essential-collection/svg/more-1.svg";
+
+export default function DropDownMenuModal(props: DropDownMenuModalProps) {
+  const scope = useCss(css);
+  const [dropDownOpen, setDropDownOpen] = React.useState(false);
+
+  const mapMenuButtons = props.buttonData.map((button, i) => {
+    return (
+      <li onClick={button.buttonAction} key={`dropdown-menu-button-${i}`}>
+        {button.buttonText}
+      </li>
+    );
+  });
+
+  return (
+    <>
+      <div className="dropdown-container" {...scope}>
+        <img
+          alt="kabob icon - drop down menu"
+          className="kabob-icon"
+          onClick={() => setDropDownOpen(!dropDownOpen)}
+          src={kabobIcon}
+        />
+        {dropDownOpen === true && (
+          <div
+            className="popup dropdown-dialog-box"
+            onClick={() => setDropDownOpen(!dropDownOpen)}
+          >
+            <ul>{mapMenuButtons}</ul>
+          </div>
+        )}
+      </div>
+      {dropDownOpen === true && (
+        <div
+          className="dropdown-modal-screen"
+          onClick={() => setDropDownOpen(!dropDownOpen)}
+          {...scope}
+        />
+      )}
+    </>
+  );
+}
+
+const css = `
+& .kabob-icon {
+	height: 2rem;
```

Comment:
Please configure your editor globally to use spaces instead of tabs. Also to use 2 spaces instead of 4. This usually doesn't matter because prettier catches it, but because template strings are whitespace sensitive it does matter for this.

https://stackoverflow.com/questions/34174207/how-to-change-indentation-in-visual-studio-code/45671704